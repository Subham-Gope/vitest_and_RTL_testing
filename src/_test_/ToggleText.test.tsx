// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import ToggleText from "../components/ToggleText";

// describe("Testing the ToggleText.tsx component", () => {
//   test("lets user toggle state", async () => {
//     const { container } = render(<ToggleText />);
//     expect(container).toMatchSnapshot();

//     const btn = screen.getByRole("button", { name: /toggle/i });
//     expect(btn).toBeInTheDocument();

//     await userEvent.click(btn);

//     const message = screen.getByText("Toggled!");

//     expect(message).toBeInTheDocument();
//   });
// });
