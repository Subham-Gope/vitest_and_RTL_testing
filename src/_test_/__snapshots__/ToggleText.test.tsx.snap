// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the ToggleText.tsx component > lets user toggle state 1`] = `
<div>
  <button>
    Toggle
  </button>
</div>
`;
